[PE-D][Tester B] Editing data file to invalid rating causes crash

## Description
If the persisted data file contains an out-of-range rating (e.g., 6), the app attempts to construct a Rating directly during deserialization, which throws an unchecked IllegalArgumentException. This can crash the load path instead of reporting a validation error or recovering gracefully as suggested by the User Guide.

## Steps to reproduce
Run the app once to generate data/foodtrail.json.
Close the app.
Open data/foodtrail.json and set any restaurant’s "rating" to 6 (or -1).
Re-launch the app.

## Expected
The app should detect invalid data and handle it gracefully:
Either reject only the invalid entry and continue, or
Reset to empty data with a clear message, consistent with the User Guide’s guidance on invalid data files.
No unhandled exceptions.

## Actual
App does not start up at all. The following exception is unhandled:

```
Exception in Application init method
Exception in thread "main" java.lang.RuntimeException: Exception in Application init method
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:896)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:196)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalArgumentException: Rating must be an integer between 0 and 5 (inclusive).
        at foodtrail.model.restaurant.Rating.<init>(Rating.java:23)
        at foodtrail.storage.JsonAdaptedRestaurant.toModelType(JsonAdaptedRestaurant.java:109)
        at foodtrail.storage.JsonSerializableRestaurantDirectory.toModelType(JsonSerializableRestaurantDirectory.java:52)
        at foodtrail.storage.JsonRestaurantDirectoryStorage.readRestaurantDirectory(JsonRestaurantDirectoryStorage.java:55)
        at foodtrail.storage.StorageManager.readRestaurantDirectory(StorageManager.java:64)
        at foodtrail.storage.StorageManager.readRestaurantDirectory(StorageManager.java:58)
        at foodtrail.MainApp.initModelManager(MainApp.java:84)
        at foodtrail.MainApp.init(MainApp.java:64)
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:825)
        ... 2 more
```

## Suggested fix

In JsonAdaptedRestaurant.toModelType():
Wrap new `Rating(rating)` in a try/catch and rethrow as `IllegalValueException` with `Rating.MESSAGE_CONSTRAINTS`, or
Validate the integer bound first and create `Optional.empty()` or fail gracefully with a handled exception.
Ensure the upper-level storage loader catches `IllegalValueException` and follows the UG behavior (e.g., reset data and continue running with a clear error message).

-------------
Labels: `severity.High` `type.FunctionalityBug`

--- a/src/main/java/foodtrail/storage/JsonAdaptedRestaurant.java
+++ b/src/main/java/foodtrail/storage/JsonAdaptedRestaurant.java
@@ -105,8 +105,16 @@ class JsonAdaptedRestaurant {
 
         final Set<Tag> modelTags = new HashSet<>(restaurantTags);
 
-        final Optional<Rating> modelRating = (rating == null) ? java.util.Optional.empty()
-                : java.util.Optional.of(new Rating(rating));
+        final Optional<Rating> modelRating;
+        if (rating == null) {
+            modelRating = Optional.empty();
+        } else {
+            try {
+                modelRating = Optional.of(new Rating(rating));
+            } catch (IllegalArgumentException e) {
+                throw new IllegalValueException(Rating.MESSAGE_CONSTRAINTS, e);
+            }
+        }
 
         final IsMarked modelIsMarked = new IsMarked(isMarked != null && isMarked);
 

--- a/src/test/java/foodtrail/storage/JsonAdaptedRestaurantTest.java
+++ b/src/test/java/foodtrail/storage/JsonAdaptedRestaurantTest.java
@@ -102,17 +102,17 @@ public class JsonAdaptedRestaurantTest {
     }
 
     @Test
-    public void toModelType_invalidRatingNegative_throwsIllegalArgumentException() {
+    public void toModelType_invalidRatingNegative_throwsIllegalValueException() {
         JsonAdaptedRestaurant restaurant =
                 new JsonAdaptedRestaurant(VALID_NAME, VALID_PHONE, VALID_ADDRESS, VALID_TAGS, -1, VALID_IS_MARKED);
-        assertThrows(IllegalArgumentException.class, Rating.MESSAGE_CONSTRAINTS, restaurant::toModelType);
+        assertThrows(IllegalValueException.class, Rating.MESSAGE_CONSTRAINTS, restaurant::toModelType);
     }
 
     @Test
-    public void toModelType_invalidRatingTooLarge_throwsIllegalArgumentException() {
+    public void toModelType_invalidRatingTooLarge_throwsIllegalValueException() {
         JsonAdaptedRestaurant restaurant =
                 new JsonAdaptedRestaurant(VALID_NAME, VALID_PHONE, VALID_ADDRESS, VALID_TAGS, 10, VALID_IS_MARKED);
-        assertThrows(IllegalArgumentException.class, Rating.MESSAGE_CONSTRAINTS, restaurant::toModelType);
+        assertThrows(IllegalValueException.class, Rating.MESSAGE_CONSTRAINTS, restaurant::toModelType);
     }
 
     @Test

